packages/web3-eth-contract/src/contract.ts

```diff
@@ -1082,19 +1073,26 @@ export class Contract<Abi extends ContractAbi>
 			} else {
 				const errors: Web3ValidationErrorObject[] = [];
 
+				// all the methods that have is valid for the given inputs
+				const applicableMethodAbi: AbiFunctionFragment[] = [];
 				for (const _abi of arrayOfAbis) {
 					try {
 						abiParams = this._getAbiParams(_abi, params);
-						validator.validate(
-							_abi.inputs as unknown as ValidationSchemaInput,
-							abiParams,
-						);
-						methodAbi = _abi;
-						break;
+						validator.validate(_abi.inputs as unknown as ValidationSchemaInput, abiParams);
+						applicableMethodAbi.push(_abi);
 					} catch (e) {
 						errors.push(e as Web3ValidationErrorObject);
 					}
 				}
+				if (applicableMethodAbi.length === 1) {
+					[methodAbi] = applicableMethodAbi; // take the first item that is the only item in the array
+				} else {
+					[methodAbi] = applicableMethodAbi; // take the first item in the array
+					// TODO: Should throw a new error with the list of methods found.
```

Comment:
An issue has been created: https://github.com/web3/web3.js/issues/6923
The changes was not applied to not introduce a breaking change. If someone things this is an acceptable change in 4.x. We can discuss and probably implement.